### DISCUSSION TITLE
New Shape Generation Feature for draw.io

### DISCUSSION THREAD

**Discussion by RoccoMarco** [jgraph/drawio#3463] [Ideas]
Dear draw.io Team,

I am an avid user of Draw.io and have come up with a suggestion that would increase productivity with the tool: Shape Generation. The feature would users to create custom shapes by combining simple ones in various ways.

The 3 pillars of the ideas are:
1. Chance to combine shapes with multiple methods: it allows users to combine basic shapes using a variety of methods, including addition, subtraction, and intersection. This would enable the creation of infinite new shapes by combining the existing ones.
2. Single Shape Output: Once the new shape is created, it should become a single, unified shape that cannot be reverted to its previous state. This is different from the current "group" feature, as it would create a truly new shape to which you could apply easily effects like shadow and fill or simply resize it without messing up.
3. Custom Libraries: Once you create a shape you would like to save and recall it easily. Ideally, the feature would ideally include an intuitive UI that allows users to create a new custom library and add shapes to it. Additionally, these libraries should be easy to be exported and imported fostering collaboration and enhancing the overall draw.io experience.

I believe this feature may increase the use cases of draw.io and increase user productivity. 
Thanks,
Rocco

**Comment by davidjgraph** [comment]
I have to ask, did chatgpt write this for you?

**Reply by RoccoMarco** [reply]
I have to confess, I wrote the main points and I ask chatGPT to revise and formalize them. I hope this doesn't make the idea less valuable.

**Comment by sanksara** [comment]
It is strange.. that such a tool doesnot exist.. and these people only want to use the standard library.
i spent a whole day trying to create a combined cl;osed path shape. It seems it is not possible

**Comment by chryss** [comment]
I'm currently evaluating draw.io for my purposes (specifically, concept diagrams to support scientific collaborations and idea generation), because ... Powerpoint is currently the best thing we have (srsly). What I'm currently hung up at is that draw.io doesn't support shape operations (union, difference, and "fragment") that are absolutely critical for an efficient workflow. 

So while I'm not sure this is exactly what the OP requested, there's clearly some overlap in what I and my coworkers need.
